src/index.cpp

```diff
@@ -4395,6 +4396,28 @@ void Index::compute_sort_scores(const std::vector<sort_by>& sort_fields, const i
             scores[0] = int64_t(found);
         } else if(field_values[0] == &vector_distance_sentinel_value) {
             scores[0] = float_to_int64_t(vector_distance);
+        } else if(field_values[0] == &vector_query_sentinel_value) {
+            scores[0] = float_to_int64_t(2.0f);
+            try {
+                auto& field_vector_index = vector_index.at(sort_fields[0].vector_query.field_name);
```

Comment:
Should avoid this lookup here. Let's store `field_vector_index` as part of sort_fields by wrapping vector_query by another struct.